src/main/java/com/microsoft/azure/vmagent/AzureVMManagementServiceDelegate.java

```diff
@@ -227,10 +225,10 @@ public AzureVMDeploymentInfo createDeployment(
             String blobEndpointSuffix = getBlobEndpointSuffixForTemplate(storageAccount);
 
 
-            Boolean isBasic = template.isTopLevelType(Constants.IMAGE_TOP_LEVEL_BASIC);
-            ImageReferenceType referenceType = ImageReferenceType.get(template.getImageReferenceType());
+            boolean isBasic = template.isTopLevelType(Constants.IMAGE_TOP_LEVEL_BASIC);
+            ImageReferenceType referenceType = template.getImageReference().getType();
```

Comment:
This also may throw NPE since `readResolve` doese not init it properly.

Reply:
how doesn't it init it properly?
https://github.com/timja/azure-vm-agents-plugin/blob/0d4055889f6c1e2f14a73bdb710fbd2661349bc3/src/main/java/com/microsoft/azure/vmagent/AzureVMAgentTemplate.java#L746

Reply:
I assume you meant this:
https://github.com/jenkinsci/azure-vm-agents-plugin/pull/144#discussion_r281587858